Packs/Jira/Integrations/JiraV3/JiraV3_test.py

```diff
@@ -1497,6 +1497,114 @@ def test_upload_file_command(self, mocker):
         command_results = upload_file_command(client=client, args={'issue_key': 'COMPANYSA-35'})
         assert command_results.to_context()['HumanReadable'] == expected_command_results_context['HumanReadable']
 
+    def test_upload_XSOAR_attachment_to_jira_mime_type_check(self, mocker):
+        """
+        Given:
+            - A Jira client.
+        When
+            - When calling the jira-issue-upload-file command.
+        Then
+            - Validate that correct mime_type was given to the file.
+        """
+        from JiraV3 import upload_XSOAR_attachment_to_jira
+        client = jira_base_client_mock()
+        file_name = 'dummy_file_name.pdf'
+        issue_key = 'COMPANYSA-35'
+        file_bytes = b'dummy content'
+        expected_file_mime_type = 'application/pdf'
+        upload_file_raw_response = util_load_json('test_data/upload_file_test/raw_response.json')
+        files = {'file': (file_name, file_bytes, expected_file_mime_type)}
+        mocker.patch('JiraV3.get_file_name_and_content', return_value=('dummy_file_name.pdf', b'dummy content'))
+        mocker.patch('JiraV3.guess_type', return_value=(expected_file_mime_type, ''))
+        mock_request = mocker.patch.object(client, 'upload_attachment', return_value=upload_file_raw_response)
+        upload_XSOAR_attachment_to_jira(client=client,
+                                        entry_id='',
+                                        issue_id_or_key=issue_key)
+        mock_request.assert_called_with(issue_id_or_key=issue_key,
+                                        files=files)
+
+    def test_upload_XSOAR_attachment_to_jira_mime_type_fail(self, mocker):
+        """
+        Given:
+            - A Jira client.
+        When
+            - When calling the jira-issue-upload-file command.
+        Then
+            - Validate that in case of unsuccessful upload to Jira due to mime type issue,
+            we will try again with the default mime type.
+        """
+        from JiraV3 import upload_XSOAR_attachment_to_jira
+        client = jira_base_client_mock()
+        issue_key = 'COMPANYSA-35'
+        expected_file_mime_type = 'application/pdf'
+        mocker.patch('JiraV3.get_file_name_and_content', return_value=('dummy_file_name.pdf', b'dummy content'))
+        mocker.patch('JiraV3.guess_type', return_value=(expected_file_mime_type, ''))
+        mocker.patch.object(client, 'upload_attachment', side_effect=DemistoException('failed to upload', res={}))
+        mock_demisto = mocker.patch('JiraV3.demisto')
+        try:
+            upload_XSOAR_attachment_to_jira(client=client, entry_id='', issue_id_or_key=issue_key)
+        except Exception:
+            mock_demisto.debug.assert_any_call("The first call to upload_attachment() with "
+                                               "chosen_file_mime_type='application/pdf' failed. Trying again with "
+                                               "file_mime_type=application-type")
```

Comment:
We mock the two runs of the upload_attachment method, and check the input of the second time.
```suggestion
        from JiraV3 import upload_XSOAR_attachment_to_jira
        client = jira_base_client_mock()
        issue_key = 'COMPANYSA-35'
        mocker.patch('JiraV3.get_file_name_and_content', return_value=('dummy_file_name.pdf', b'dummy content'))
        mocker.patch('JiraV3.guess_type', return_value=('application/pdf', ''))
        mock_request = mocker.patch.object(client, 'upload_attachment',
                                           side_effect=[DemistoException('failed to upload', res={}), {}])
        mock_demisto = mocker.patch('JiraV3.demisto')
        upload_XSOAR_attachment_to_jira(client=client, entry_id='', issue_id_or_key=issue_key)
        # Validate that we run upload_attachment twice, once with an error, and second time to use default file type
        assert mock_request.call_count == 2
        # Validate that the second call uses the default file type (application-type)
        mock_request.assert_called_with(files={'file': ('dummy_file_name.pdf', b'dummy content', 'application-type')},
                                        issue_id_or_key=issue_key)
        mock_demisto.debug.assert_any_call("The first call to upload_attachment() with "
                                            "chosen_file_mime_type='application/pdf' failed. Trying again with "
                                            "file_mime_type=application-type")
```